### ISSUE TITLE
train a new model with additional "challenging images" annotation data

### ISSUE THREAD

**Issue by keighrim** [clamsproject/app-swt-detection#116]
### Because

With the additional data (https://github.com/clamsproject/aapb-annotations/pull/98) and new `data_loader.py` code (#115), I'd like to conduct experiments with new models and see if (and how much) the additional data is helpful improving the classification performance. 

### Done when

- [x] New models with different hyper parameters are trained and evaluated ~(via k-fold validation)~
- [x] ~The k-fold validation results are analyzed to measure the impact of the new data~ (no longer doing k-fold, see below)
- [x] If necessary, release a new version of SWT app with newly trained models

**Comment by keighrim** [comment]
Recent conversation added some changes to the plan. Specifically, in addition to the existing k-fold validation during training, we'd like to build a new set of fixed validation set for conventional training-validation workflow. That new set will be based on the `2-X_pbd` subset in the "challenging images" dataset. That leaves only `1-1_bm` to be added to the training. 

Regarding the "impact" of the new data, we'd also like to use the fixed validation set for different models from different subsets of training data, that can show the impact of incremental annotations over time in the past year or so.

**Comment by keighrim** [comment]
copying a message from @owencking over slack today.

---
> I finished creating that ["PBD" evaluation set](https://drive.google.com/file/d/1LO8yCQtWkBiu2pihh-iKvb1EevgONgH6/view?usp=drive_link).  The linked file includes images, labels, and some documentation.
There is no overlap between the assets in this set and the assets used in any training set.  Moreover, there is no (or minimal) overlap between the programs/series in this set and those in the training sets.  Hence, this set can serve as a benchmark test set across rounds of training.  Furthermore, it can serve as a check against overfitting.

**Comment by keighrim** [comment]
I ended up with running a set of experiments using the same hyperparameters from the latest version, but adding other convnext backbones with different sizes as Owen requested. For this experiment, my primary goal was to see the impact of the training data size. 

Here's the plotted results - 
[exp-trainsize.zip](https://github.com/user-attachments/files/17531748/exp-trainsize.zip)

In the plots, `Overall` shows the macro average of P/R/F scores across all 19 raw labels (no subtypes), and `all` shows individual P/R/F scores for select raw labels.

**Comment by keighrim** [comment]
After finding out about parallel coordinates visualization (https://github.com/clamsproject/aapb-evaluations/issues/60#issuecomment-2452444663), I ran another round of grid search with a full(er) hyperparameter values. The results visualized in parallel coordinates with different label aggregations are available in [swt-7.1rc-res-gridsearch-results.tar.gz](https://github.com/user-attachments/files/17896088/swt-7.1rc-res-gridsearch-results.tar.gz). 
Some findings from the analyzing the results 

- Deeper network (more layers) didn’t help
  - Model easily goes "overfitting"
- When the network is shallow, longer training didn’t help a lot (still helpful)
- With long-enough trained shallow network, positional encoding showed big differences in slates and credits
- On average, the larger training data, the better performance
  - still, even with the maximum available training data, label distribution is severely skewed

Full analysis report is available at https://docs.google.com/presentation/d/14z1NrZcamRVOn5H6N2e3jjnfu8u1140XL7mhuDKrEtY/edit?usp=sharing (access restricted to the team members)

**Comment by keighrim** [comment]
626171c00d3eba16ab11e77db15c175cb110a2da added newly trained models. Some notable changes in new models: 

- label `U` is added - this is a label that's newly introduced during the annotation of the "challenging" images (6/21 edits of [the label definitions](https://docs.google.com/document/d/1IyM_rCsCr_1XQ39j36WMX-XnVVBT4T_01j-M0eYqyDs/edit?usp=sharing).
- in additional to `convnext_lg` and `convnext_tiny`, `convnext_small`-based models are added. The default is now `convnext_small` model.

**Comment by keighrim** [comment]
Reopening the issue for re-training and re-release of a model that includes the ["confirmed-slates" annotations](https://github.com/clamsproject/aapb-collaboration/issues/27#issuecomment-2501578284) in the training data.

**Comment by keighrim** [comment]
Some reports from the newly trained models with "confirmed-slates" data. This round of training experiment is done with the hyperparameter values found in the previous "7.1rc" experiment (reported in https://docs.google.com/presentation/d/14z1NrZcamRVOn5H6N2e3jjnfu8u1140XL7mhuDKrEtY/edit?usp=sharing ), but just varying the training data size between `all-available` (58250 images) vs. `all-by-confirmed-slates` (57134 images). 

1. with positional encoding
    - overall average showed no practical difference between different sized backbones and different sized training data.
    - across all sizes of backbones, adding 1116 more slate images showed performance degradation for `O`, `M`, `L`, `E` labels, but showed some clear gains for `S` and `G` labels. 
2. without positional encoding
    - again, overall average showed no significant differences.  
    - larger training data showed clear gain for `S`, minor gain for `L` and `G`, but loss for `M`. 
  
Attaching parallel coordinate visualization I used for analysis. 
[results.confirmedSlates-gridsearch-results.zip](https://github.com/user-attachments/files/17979250/results.confirmedSlates-gridsearch-results.zip)

I will release 7.1 with the newly trained model (with confirmed-slates annotations), as it didn't show any catastrophic performance degradation across all sized backbones.

**Comment by keighrim** [comment]
closed via `v7.1` release (https://github.com/clamsproject/app-swt-detection/tree/v7.1).
